File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
	if len(prices) == 0 {
		return 0
	}
	minPrice := prices[0]
	maxProf := 0

	for i := 1; i < len(prices); i++ {
		if prices[i] < minPrice {
			minPrice = prices[i]
		} else if prices[i] - minPrice > maxProf {
			maxProf = prices[i] - minPrice
		}
	}
	return maxProf
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let minPrice = prices[0];
    let max = 0;
    for (let i of prices) {
        if (i < minPrice) {
            minPrice = i;
        }
        else if (i - minPrice > max) {
            max = i - minPrice;
        }
    }
    return max;
};